src/main/kotlin/auth/AuthProvderAtom.kt

```diff
@@ -0,0 +1,14 @@
+package se.zensum.leia.auth
+
+import leia.Atom
+import leia.logic.IncomingRequest
+import se.zensum.leia.auth.AuthProvider
+
+class AuthProviderAtom(private var currentProvider: AuthProvider): AuthProvider, Atom<AuthProvider> {
```

Comment:
Hm, yes, unless you remember having some special reason for not using it. But it might also be out of scope for this PR, maybe we can do it afterwards instead (unless you insist).

Reply:
Yes lets do it for all the atoms in one go